torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1098,6 +1098,16 @@ def parallel_and(*args: BoolLikeType) -> BoolLikeType:
     return all(args)
 
 
+def parallel_and_statically_known_true(*args: BoolLikeType) -> BoolLikeType:
+    """
+    Like parallel_and except that it does not guard on unbacked SymInts and may return False
+    even if the whole expression is True in some cases.
+    """
+    if any(statically_known_true(torch.sym_not(a)) for a in args):
+        return False
+    return all(args)
```

Comment:
By in large, we are not using `parallel_and` or `parallel_or` (because we have mostly solved these problems in other ways, e.g., 0/1 specialization), so I'm not that hot on adding this new variation on the function. Actually honestly we should probably remove the old ones.

Reply:
I see I can just use statically_known_true directly there and remove this abstraction.

Reply:
https://github.com/pytorch/pytorch/pull/138135

Reply:
I got red of it , but now looking at its really not probably what i originally intended. 
I wanted all(statically_known_true(a)) for a in args) anyhow i inlined it up there.